### PR TITLE
refactor(cdnjs): Use zod schema and `Result` class

### PR DESCRIPTION
**Pull request by zharinov**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/cdnjs/index.ts [hunk 1]
```diff
@@ -1,7 +1,30 @@
...
+  .optional()
+  .catch(undefined);
+
+const Assets = z.array(
+  z.object({
+    version: z.string(),
+    files: z.string().array(),
+    sri: z.record(z.string()).optional(),
+  })
+);
```
Comment (viceice) [inline lib/modules/datasource/cdnjs/index.ts:27]: why not moving those to schema.ts too?

**Review by viceice — COMMENTED** [review]
strange devcontainer failure 😕

File: lib/modules/datasource/cdnjs/index.ts (hunk 1 above)
Comment (zharinov) [reply lib/modules/datasource/cdnjs/index.ts:27]: The code for this datasource is not too big, so I think it's better to contain all of it in the single file

Comment (viceice) [reply lib/modules/datasource/cdnjs/index.ts:27]: then also inline the one zod schema which resides in schema.ts

Comment (zharinov) [reply lib/modules/datasource/cdnjs/index.ts:27]: But there is no `schema.ts`

Comment (viceice) [reply lib/modules/datasource/cdnjs/index.ts:27]: 🙈 sorry, I misread the datasource general schema.ts 😅

**Review by viceice — APPROVED** [review]

**Pull request merged**